test/runtests.jl

```diff
@@ -44,61 +45,70 @@ setprogress!(false)
+macro timeit_include(path::AbstractString)
+    return :(@timeit TIMEROUTPUT $path include($path))
+end
+
 @testset "Turing" begin
     @testset "essential" begin
-        include("essential/ad.jl")
+        @timeit_include("essential/ad.jl")
     end
 
     @testset "samplers (without AD)" begin
-        include("inference/AdvancedSMC.jl")
-        include("inference/emcee.jl")
-        include("inference/ess.jl")
-        include("inference/is.jl")
+        @timeit_include("inference/AdvancedSMC.jl")
+        @timeit_include("inference/emcee.jl")
+        @timeit_include("inference/ess.jl")
+        @timeit_include("inference/is.jl")
     end
 
     Turing.setrdcache(false)
     for adbackend in (:forwarddiff, :tracker, :reversediff)
-        Turing.setadbackend(adbackend)
-        @info "Testing $(adbackend)"
-        start = time()
-        @testset "inference: $adbackend" begin
-            @testset "samplers" begin
-                include("inference/gibbs.jl")
-                include("inference/gibbs_conditional.jl")
-                include("inference/hmc.jl")
-                include("inference/Inference.jl")
-                include("contrib/inference/dynamichmc.jl")
-                include("contrib/inference/sghmc.jl")
-                include("inference/mh.jl")
+        @timeit TIMEROUTPUT "inference: $adbackend" begin
+            Turing.setadbackend(adbackend)
+            @info "Testing $(adbackend)"
+            start = time()
+            @testset "inference: $adbackend" begin
+                @testset "samplers" begin
+                    @timeit_include("inference/gibbs.jl")
+                    @timeit_include("inference/gibbs_conditional.jl")
+                    @timeit_include("inference/hmc.jl")
+                    @timeit_include("inference/Inference.jl")
+                    @timeit_include("contrib/inference/dynamichmc.jl")
+                    @timeit_include("contrib/inference/sghmc.jl")
+                    @timeit_include("inference/mh.jl")
+                end
             end
-        end
 
-        @testset "variational algorithms : $adbackend" begin
-            include("variational/advi.jl")
-        end
+            @testset "variational algorithms : $adbackend" begin
+                @timeit_include("variational/advi.jl")
+            end
 
-        @testset "modes" begin
-            include("modes/ModeEstimation.jl")
-            include("modes/OptimInterface.jl")
-        end
+            @testset "modes : $adbackend" begin
+                @timeit_include("modes/ModeEstimation.jl")
+                @timeit_include("modes/OptimInterface.jl")
+            end
 
-        # Useful for
-        # a) discovering performance regressions,
-        # b) figuring out why CI is timing out.
-        @info "Tests for $(adbackend) took $(time() - start) seconds"
+            # Useful for figuring out why CI is timing out.
+            @info "Tests for $(adbackend) took $(time() - start) seconds"
```

Comment:
Still needed?

```suggestion
```

Reply:
Ah yes no indeed not. I thought so, but only the `@info "Testing $(adbackend)" ` should be enough.